src/import-data/useImportRequest.ts

```diff
@@ -0,0 +1,136 @@
+import { useRoute } from 'src/libs/nav';
+
+import {
+  CatalogDatasetImportRequest,
+  CatalogSnapshotsImportRequest,
+  FileImportRequest,
+  ImportRequest,
+  TDRSnapshotExportImportRequest,
+  TDRSnapshotReferenceImportRequest,
+} from './import-types';
+
+type QueryParams = { [key: string]: unknown };
+
+/**
+ * Validate that a value is a string. Throw an error if it is not.
+ *
+ * @param value The value to validate.
+ * @param label Label for the value to use in error messages.
+ * @returns The validated string value.
+ */
+const requireString = (value: unknown, label = 'value'): string => {
```

Comment:
Is there a place for general validation functions like this?

Reply:
I suppose this could go in a `libs/validation-utils` or something like that. But I'm not sure I want to promote it as a shared utility.

We do have a dependency on [validate.js](https://www.npmjs.com/package/validate.js), but it's been archived since 2021 so we should probably figure out a different solution for validation.